docs/sphinx-docs/doc-regen/makedocumentation.py

```diff
@@ -0,0 +1,132 @@
+"""
+Creates documentation from .py files
+"""
+import os
+import sys
+from os.path import join, abspath, dirname, basename
+import subprocess
+
+FILE_ABS_SOURCE = abspath(dirname(__file__))
+MAIN_PY_SRC = FILE_ABS_SOURCE + "/../source-temp/user/models/src/"
+if os.path.exists(MAIN_PY_SRC):
+    ABSOLUTE_TARGET_MAIN = abspath(join(dirname(__file__), MAIN_PY_SRC))
+    PLUGIN_PY_SRC = abspath(dirname(__file__)) + "/../../../../.sasview/plugin_models/"
+    ABSOLUTE_TARGET_PLUGINS = abspath(join(dirname(__file__), PLUGIN_PY_SRC))
+else:
+    pass
+
+def get_py(directory):
+    for root, dirs, files in os.walk(directory):
+        # Only include python files not starting in '_' (pycache not included)
+        PY_FILES = [join(directory, string) for string in files if not string.startswith("_") and string.endswith(".py")]
+        return PY_FILES
+
+def get_main_docs():
+    """
+Generates string of .py files to be passed into compiling functions
+Future reference: move to main() function?
+    """
+    # The order in which these are added is important. if ABSOLUTE_TARGET_PLUGINS goes first, then we're not compiling the .py file stored in .sasview/plugin_models
+    TARGETS = get_py(ABSOLUTE_TARGET_MAIN) + get_py(ABSOLUTE_TARGET_PLUGINS)
```

Comment:
I believe there are some deeper concerns here that would prevent the implementation of either of these solutions for the time being. The same collision resolution used by sasview should not be applied here because we're dealing with filepath names to pass to Sphinx and not the names of the models in SasView themselves. Filepaths have to be unique, so this shouldn't be a concern. The reason we need to have collision resolution here anyways is because `regenmodel.py` copies user created files into `source-temp`, the folder that stores the copies of models we use for editing documentation in this branch, because apparently Sphinx can't link to absolute paths. We search in both directories because the user may have added new models to the user model directory before `regenmodel.py` has run, and we can't run `regenmodel.py` first because it would take too long and we need to pass in only one file. A potential solution *could* be to use Jeff's suggested approach of putting `source-temp` inside the user model directory itself, but I think that's too big of a change for the time being and would involve some more research into whether or not Jeff's idea is compatible with Sphinx linking to source files.